live.tsx

```diff
@@ -144,27 +156,95 @@ export function createLiveHandler<LoaderData = LivePageData>(
       }
 
       start("fetch-page-data");
-      let loaderData = undefined;
-
-      try {
-        if (typeof loader === "function") {
-          loaderData = await loader(req, ctx);
-        } else {
-          loaderData = await loadLiveComponents(
-            req,
-            ctx,
-            options as LoadLiveComponentsOptions,
-          );
+      const pageData = await loadLiveComponents(
+        req,
+        ctx,
+        options as LoadLiveComponentsOptions,
+      );
+      end("fetch-page-data");
+
+      const isLoaderProp = (value: any): value is string =>
+        typeof value === "string" && value.charAt(0) === "{" &&
+        value.charAt(value.length - 1) === "}";
+
+      // map back components from database to components for the editor, merging loader props into component props
+      const editorComponents = pageData.components.map((componentData) => {
+        if (
+          Object.values(componentData.props ?? {}).some(isLoaderProp)
+        ) {
```

Comment:
Do we need this `if` if you're already checking each prop in the iteration below?